include/deal.II/grid/data_transfer.h

```diff
@@ -0,0 +1,235 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 1998 - 2023 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_tria_data_transfer_h
+#define dealii_tria_data_transfer_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/mpi.h>
+#include <deal.II/base/mpi_large_count.h>
+#include <deal.II/base/utilities.h>
+
+#include <deal.II/grid/cell_status.h>
+#include <deal.II/grid/tria_iterator_selector.h>
+
+#include <boost/range/iterator_range.hpp>
+
+#include <algorithm>
+#include <fstream>
+#include <functional>
+#include <utility>
+#include <vector>
+
+DEAL_II_NAMESPACE_OPEN
+
+template <int dim, int spacedim = dim>
+DEAL_II_CXX20_REQUIRES((concepts::is_valid_dim_spacedim<dim, spacedim>))
+struct CellAttachedData
```

Comment:
Can you add documentation to this class?

I suspect that this is a class that will only be used internally, right? If so, perhaps move it to a `namespace internal`?